blurring_as_a_service/metrics/custom_metrics_calculator.py

```diff
@@ -0,0 +1,500 @@
+        """
+        Parameters
+        ----------
+        tagged_validation_folder
+            Folder containing yolo tagged validation files.
+        coco_file_with_categories
+            File containing the categories.
+        """
+
+        self._get_and_prepare_categories(coco_file_with_categories)
+        self._calculate_true_positives_and_false_negatives_per_category(
+            tagged_validation_folder
+        )
+        self._get_gt_boxes_and_predicted_boxes_per_category(tagged_validation_folder)
+
+    def calculate_and_store_metrics(self, markdown_output_path):
+        """
+        Calculates the custom metrics for all the categories and stores them in a markdown file.
+
+        Parameters
+        ----------
+        markdown_output_path
+            Path where to store the markdown file.
+
+        """
+        result = self.calculate_false_negative_rate_for_all_categories()
+
+        with open(markdown_output_path, "w") as f:
+            f.write(
+                "| Category | Value | True Positives | TP Small | TP Medium | TP Large "
+                "| False Negatives | FN Small | FN Medium | FN Large | False Negative Rate | False Negative Rate Small "
+                "|False Negative Rate Medium | False Negative Rate Large|\n"
+            )
+            f.write(
+                "| -------- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |"
+                "-------------- | --------------- | ------------------ |\n"
+            )
+            for item in result:
+                f.write(
+                    f'| {item["category"]} | {item["value"]} | {item["true_positives"]} '
+                    f'| {item["true_positives_small"]} | {item["true_positives_medium"]} | {item["true_positives_large"]} '
+                    f'| {item["false_negatives"]} | {item["false_negatives_small"]} | {item["false_negatives_medium"]} '
+                    f'| {item["false_negatives_large"]} | {item["false_negative_rate"]}'
+                    f'| {item["false_negative_rate_small"]} | {item["false_negative_rate_medium"]}'
+                    f'| {item["false_negative_rate_large"]}\n'
+                )
+            f.write(
+                f"Thresholds used for these calculations: Small=`{Size.small.value}`, Medium=`{Size.medium.value}` "
+                f"and Large=`{Size.large.value}`."
+            )
+
+    def calculate_false_negative_rate_for_all_categories(self) -> List[dict]:
+        """
+        Using the calculate_false_negative_rate_for_a_category function, it calculates the value for the following categories:
+            - 'grouped_category': Persons or licence plates.
+            - 'sex'.
+            - 'age'.
+            - 'skin_color'.
+        Returns
+        -------
+        A list where each element is a category and its metrics, included the false negative rate.
+        """
+        result = []
+        for category in [
+            "grouped_category",
```

Comment:
I know I wrote the code and it's my fault :P But since you did a nice job adding the class for the ImageSize. 
Can you do the same for the SensitiveCategories?

Reply:
Not sure what you mean here. Because the Enum makes sense for binding small, large, medium to intervals, but in this case it is just a list of strings. 

This is what I made, is this what you meant?

```
@dataclass
class SensitiveCategories:
    values: List[str] = None

    def __post_init__(self):
        if self.values is None:
            self.values = ["grouped_category", "sex", "age", "skin_color", "licence_plate_origin", "licence_plate_color"]
```

Reply:
Mmm you're right. 
I don't know what I was exactly thinking but we can leave it as it is.